pkg/roachprod/install/cluster_synced.go

```diff
@@ -1420,7 +1420,16 @@ func (c *SyncedCluster) Wait(ctx context.Context, l *logger.Logger) error {
 				return res, nil
 			}
 			res.Err = errors.Wrapf(res.Err, "timed out after 5m")
+			logContent, err := c.runCmdOnSingleNode(ctx, nil, node, fmt.Sprintf("tail -n %d %s", 20, vm.StartupLogs), cmdOptsWithDebugDisabled())
+			if err != nil {
+				res.Err = errors.CombineErrors(res.Err, err)
+			} else if logContent.Err != nil {
+				res.Err = errors.CombineErrors(res.Err, logContent.Err)
+			} else {
+				res.Err = errors.Wrapf(res.Err, "last 20 lines of log:\n%s", logContent.Stdout)
+			}
```

Comment:
Adding an error for the fetching of logs using the `CombineErrors` API is not ideal: in the current code, it won't show up at all because we're printing the error with `%v` (see [this table](https://github.com/cockroachdb/errors?tab=readme-ov-file#what-comes-out-of-an-error)). We also don't want to use `%+v` there because it would make the output very verbose.

I suggest we just _log_ the last 20 lines (or the error fetching them) instead of wrapping the error. Something like the following diff:

```diff
diff --git a/pkg/roachprod/install/cluster_synced.go b/pkg/roachprod/install/cluster_synced.go
index 49119734bb7..b2d3c43da50 100644
--- a/pkg/roachprod/install/cluster_synced.go
+++ b/pkg/roachprod/install/cluster_synced.go
@@ -1421,15 +1421,13 @@ func (c *SyncedCluster) Wait(ctx context.Context, l *logger.Logger) error {
 			}
 			res.Err = errors.Wrapf(res.Err, "timed out after 5m")
 			logContent, err := c.runCmdOnSingleNode(ctx, nil, node, fmt.Sprintf("tail -n %d %s", 20, vm.StartupLogs), cmdOptsWithDebugDisabled())
-			if err != nil {
-				res.Err = errors.CombineErrors(res.Err, err)
-			} else if logContent.Err != nil {
-				res.Err = errors.CombineErrors(res.Err, logContent.Err)
+			if err := errors.CombineErrors(err, res.Err); err != nil {
+				l.Printf("could not fetch startup logs: %v", err)
 			} else {
-				res.Err = errors.Wrapf(res.Err, "last 20 lines of log:\n%s", logContent.Stdout)
+				l.Printf("startup failed, last 20 lines of output:\n%s", logContent.CombinedOut)
+				l.Printf("  %2d: view the full log in %s", node, vm.StartupLogs)
 			}
 			l.Printf("  %2d: %v", node, res.Err)
-			l.Printf("  %2d: view the full log in %s", node, vm.StartupLogs)
 			return res, nil
 		})
```

Note also that we need to use `CombinedOut` there, not `Stdout` (a tricky part of that API).

Reply:
Thanks for the information. A few TILs.